js/plates-model/plate.js

```diff
@@ -79,6 +84,49 @@ export default class Plate extends PlateBase {
     return totalTorque.applyMatrix3(this.invMomentOfInertia)
   }
 
+  updateCenter() {
+    const safeFields = {}
+    let safeSum = new THREE.Vector3()
+    let safeArea = 0
+    this.fields.forEach(field => {
+      if (!field.subduction) {
+        let safe = true
+        // Some subducting fields do not get marked because they move so slowly
+        // Ignore fields adjacent to subducting fields just to be safe
+        field.forEachNeighbour(neighbor => {
+          if (neighbor.subduction) {
+            safe = false
+          }
+        })
+        if (safe) {
+          safeFields[field.id] = field
+          safeSum = safeSum.add(field.absolutePos)
+          safeArea += field.area
+        }
+      }
+    })
+
+    if (safeArea < 100000) {
```

Comment:
I usually try to keep such magic numbers as camel case constants at the top of the file.